### PR TITLE
Hide Verse Numbers that their corresponding verse is toggled to be hidden

### PR DESCRIPTION
**Pull request by Christopher-Shiu**
Issue #494
Problem: Verse Numbers were still present

### CODE REVIEW

File: convert/convertBooks.ts [hunk 1]
```diff
@@ -98,6 +121,31 @@ function isImageMissing(imageSource: string): boolean {
...
+//         }
+//     }
+
+//     // Call the function to update the config
+//     updateConfigForEmptyVerses(configData);
+
+//     // Check the updated configData
+//     console.log(configData);
+// });
+
```
Comment (chrisvire) [inline convert/convertBooks.ts:148]: @Christopher-Shiu Why is this here?

File: convert/convertBooks.ts [hunk 2]
```diff
@@ -57,6 +57,29 @@ function removeStrongNumberReferences(text: string, _bcId: string, _bookId: stri
...
+    // \v (any number of digits) followed by any number of such patterns
+    const regex = /(\\v\s\d+\s*)+/g;
+
+    // Replace matches with the last occurrence
+    return text.replace(regex, (match) => {
+        // Split the matched string by \v and filter out empty parts
+        const parts = match
+            .trim()
+            .split(/\\v\s*/)
+            .filter(Boolean);
```
Comment (chrisvire) [inline convert/convertBooks.ts:71]: What does `.filter(Boolean)` do?

Comment (davidmoore1) [reply convert/convertBooks.ts:71]: It filters out any empty strings

**Review by davidmoore1 — APPROVED** [review]
